### PR TITLE
Unity v2018.4 Support

### PR DESCRIPTION
**Pull request by fakegood**
- Fixed MonoHooks asmdef reference to Core asmdef was broken.
- Fixed dependancy issue for Package Manager.
- Removed Generator function in version 2018.4.

### CODE REVIEW

File: Packages/Core/Editor/Drawers/AtomDrawer.cs [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#if UNITY_2019_1_OR_NEWER
```
Comment (soraphis) [inline Packages/Core/Editor/Drawers/AtomDrawer.cs]: should be 2018_4_or_newer
instead of removing, or?

Comment (fakegood) [reply Packages/Core/Editor/Drawers/AtomDrawer.cs]: updated to have UNITY_2018.4_OR_NEWER and also updated the summary comment.

**Comment by fakegood** [comment]
> > Removed Generator function in version 2018.4.
> 
> why?

because RegenerateAllAtoms will not work properly. GeneratorEditor will need a re-write to support 2018.4 also. I will submit another PR that would have all of the Generator support. Or you prefer it to be in this PR?

**Comment by AdamRamberg** [comment]
Thanks for the PR! ❤️ It looks good! However, could you please make the PR against the canary branch instead? (they are on the same commit so should be easy) After that I will merge this one. 

You can add an issue + create a new PR for making RegenerateAllAtoms + GeneratorEditor support 2018.4.

**Comment by soraphis** [comment]
@AdamRamberg i changed the base to canary

**Pull request merged**